packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,539 @@
+      txid: input.hash.reverse().toString("hex"),
+      vout: input.index,
+      sequence: input.sequence,
+      scriptSig: input.script.toString("hex"),
+      witness: input.witness.map((w) => w.toString("hex")),
+    }));
+  }
+
+  /**
+   * Analyzes the outputs of the transaction.
+   * @returns {TransactionOutput[]} An array of analyzed transaction outputs
+   * @protected
+   */
+  protected analyzeOutputs(): TransactionOutput[] {
+    return this.originalTx.outs.map((output, index) => ({
+      value: output.value,
+      scriptPubKey: output.script.toString("hex"),
+      address: this.getOutputAddress(output.script),
+      isChange: index === this.changeOutputIndex,
+    }));
+  }
+
+  /**
+   * Attempts to derive the address from an output script.
+   * @param {Buffer} script - The output script
+   * @returns {string} The derived address or an error message if unable to derive
+   * @protected
+   */
+  protected getOutputAddress(script: Buffer): string {
+    try {
+      const p2pkhAddress = payments.p2pkh({
+        output: script,
+        network: networkData(this.network),
+      }).address;
+      const p2shAddress = payments.p2sh({
+        output: script,
+        network: networkData(this.network),
+      }).address;
+      const p2wpkhAddress = payments.p2wpkh({
+        output: script,
+        network: networkData(this.network),
+      }).address;
+      const p2wshAddress = payments.p2wsh({
+        output: script,
+        network: networkData(this.network),
+      }).address;
+
+      return (
+        p2pkhAddress ||
+        p2shAddress ||
+        p2wpkhAddress ||
+        p2wshAddress ||
+        "Unable to derive address"
+      );
+    } catch (e) {
+      return "Unable to derive address";
+    }
+  }
+
+  /**
+   * Calculates the fee for the transaction.
+   * @returns {BigNumber} The calculated fee in satoshis
+   * @protected
+   */
+  protected calculateFee(): BigNumber {
+    return new BigNumber(this.currentFeeRate).multipliedBy(this.vsize);
+  }
+
+  /**
+   * Checks if the transaction signals RBF (Replace-By-Fee).
+   * @returns {boolean} True if the transaction signals RBF, false otherwise
+   * @protected
+   */
+  protected isRBFSignaled(): boolean {
+    return this.originalTx.ins.some((input) => input.sequence < 0xfffffffe);
+  }
+
+  /**
+   * Determines if RBF (Replace-By-Fee) can be performed on this transaction.
+   * @returns {boolean} True if RBF can be performed, false otherwise
+   * @protected
+   */
+  protected canPerformRBF(): boolean {
+    return this.isRBFSignaled();
+  }
+
+  /**
+   * Determines if CPFP (Child-Pays-For-Parent) can be performed on this transaction.
+   * @returns {boolean} True if CPFP can be performed, false otherwise
+   * @protected
+   */
+  protected canPerformCPFP(): boolean {
+    return this.outputs.some(
+      (output) =>
+        !output.isChange &&
+        new BigNumber(output.value).isGreaterThan(this.dustThreshold),
```

Comment:
I mentioned this in a previous review, this is irrelevant to whether or not you can perform CPFP. All you need is a change output. 

Remember CPFP fee bumps by immediately spending one of the outputs of a pending transaction. An output that you can spend from another transaction is the equivalent of a change output. If I sent bitcoin to you and there are two outputs, one to you and one back to me as change, I can CPFP by spending the change output. Similarly you can CPFP by spending the output that's to you (so by this definition "change" is any output of a tx you control, so we could instead call the change output "spendable output" if we wanted).

Reply:
This stands corrected now ...

```js
  protected canPerformCPFP(): boolean {
    return this.outputs.some((output) => output.isSpendable);
  }
  ```